src/domain.jl

```diff
@@ -0,0 +1,104 @@
+abstract type Domain{N,T} end
+
+# Without S, add_set! and remove_set! where not type-stable...
+struct DomainList{N,T,S<:Grid{N,T}} <: Domain{N,T}
+    grid::S
+    elems::Set{NTuple{N,Int}}
+end
+
+function NewDomainList(grid::S) where {N,T,S<:Grid{N,T}}
```

Comment:
Why not create a classical constructor `DomainList(grid::S)` ?

Reply:
changed